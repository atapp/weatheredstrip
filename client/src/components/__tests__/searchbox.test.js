import React from 'react';
import { render, cleanup } from 'react-testing-library'
import ReactDOM from 'react-dom';
import Searchbox from '../searchbox.react';
import 'jest-dom/extend-expect';
afterEach(cleanup)

it('searchbox renders properly', () => {
  const { container } = render(<Searchbox />)
  const searchbox = container.firstChild
  expect(searchbox).toMatchSnapshot()
});

// Need function testing for form submit.
