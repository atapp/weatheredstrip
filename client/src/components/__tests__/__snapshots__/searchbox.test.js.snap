// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`searchbox data entry changes state 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SearchBox />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <input
          className="searchbox-input"
          onChange={[Function]}
          placeholder="CYMX CYUL..."
          type="text"
          value=""
        />,
        <withRouter(LinkButton)
          className="searchbox-button"
          onClick={[Function]}
          to={
            Object {
              "pathname": "/airports",
              "search": "?stations=",
            }
          }
        >
          Get Info
        </withRouter(LinkButton)>,
      ],
      "className": "searchbox",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "searchbox-input",
          "onChange": [Function],
          "placeholder": "CYMX CYUL...",
          "type": "text",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": "Get Info",
          "className": "searchbox-button",
          "onClick": [Function],
          "to": Object {
            "pathname": "/airports",
            "search": "?stations=",
          },
        },
        "ref": null,
        "rendered": "Get Info",
        "type": [Function],
      },
    ],
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <input
            className="searchbox-input"
            onChange={[Function]}
            placeholder="CYMX CYUL..."
            type="text"
            value=""
          />,
          <withRouter(LinkButton)
            className="searchbox-button"
            onClick={[Function]}
            to={
              Object {
                "pathname": "/airports",
                "search": "?stations=",
              }
            }
          >
            Get Info
          </withRouter(LinkButton)>,
        ],
        "className": "searchbox",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "searchbox-input",
            "onChange": [Function],
            "placeholder": "CYMX CYUL...",
            "type": "text",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "Get Info",
            "className": "searchbox-button",
            "onClick": [Function],
            "to": Object {
              "pathname": "/airports",
              "search": "?stations=",
            },
          },
          "ref": null,
          "rendered": "Get Info",
          "type": [Function],
        },
      ],
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`searchbox data submit 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SearchBox
    searchSubmit={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <input
          className="searchbox-input"
          onChange={[Function]}
          placeholder="CYMX CYUL..."
          type="text"
          value=""
        />,
        <withRouter(LinkButton)
          className="searchbox-button"
          onClick={[Function]}
          to={
            Object {
              "pathname": "/airports",
              "search": "?stations=",
            }
          }
        >
          Get Info
        </withRouter(LinkButton)>,
      ],
      "className": "searchbox",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "searchbox-input",
          "onChange": [Function],
          "placeholder": "CYMX CYUL...",
          "type": "text",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": "Get Info",
          "className": "searchbox-button",
          "onClick": [Function],
          "to": Object {
            "pathname": "/airports",
            "search": "?stations=",
          },
        },
        "ref": null,
        "rendered": "Get Info",
        "type": [Function],
      },
    ],
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <input
            className="searchbox-input"
            onChange={[Function]}
            placeholder="CYMX CYUL..."
            type="text"
            value=""
          />,
          <withRouter(LinkButton)
            className="searchbox-button"
            onClick={[Function]}
            to={
              Object {
                "pathname": "/airports",
                "search": "?stations=",
              }
            }
          >
            Get Info
          </withRouter(LinkButton)>,
        ],
        "className": "searchbox",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "searchbox-input",
            "onChange": [Function],
            "placeholder": "CYMX CYUL...",
            "type": "text",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "Get Info",
            "className": "searchbox-button",
            "onClick": [Function],
            "to": Object {
              "pathname": "/airports",
              "search": "?stations=",
            },
          },
          "ref": null,
          "rendered": "Get Info",
          "type": [Function],
        },
      ],
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`searchbox renders properly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SearchBox />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <input
          className="searchbox-input"
          onChange={[Function]}
          placeholder="CYMX CYUL..."
          type="text"
          value=""
        />,
        <withRouter(LinkButton)
          className="searchbox-button"
          onClick={[Function]}
          to={
            Object {
              "pathname": "/airports",
              "search": "?stations=",
            }
          }
        >
          Get Info
        </withRouter(LinkButton)>,
      ],
      "className": "searchbox",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "searchbox-input",
          "onChange": [Function],
          "placeholder": "CYMX CYUL...",
          "type": "text",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": "Get Info",
          "className": "searchbox-button",
          "onClick": [Function],
          "to": Object {
            "pathname": "/airports",
            "search": "?stations=",
          },
        },
        "ref": null,
        "rendered": "Get Info",
        "type": [Function],
      },
    ],
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <input
            className="searchbox-input"
            onChange={[Function]}
            placeholder="CYMX CYUL..."
            type="text"
            value=""
          />,
          <withRouter(LinkButton)
            className="searchbox-button"
            onClick={[Function]}
            to={
              Object {
                "pathname": "/airports",
                "search": "?stations=",
              }
            }
          >
            Get Info
          </withRouter(LinkButton)>,
        ],
        "className": "searchbox",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "searchbox-input",
            "onChange": [Function],
            "placeholder": "CYMX CYUL...",
            "type": "text",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "Get Info",
            "className": "searchbox-button",
            "onClick": [Function],
            "to": Object {
              "pathname": "/airports",
              "search": "?stations=",
            },
          },
          "ref": null,
          "rendered": "Get Info",
          "type": [Function],
        },
      ],
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`searchbutton report refresh 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SearchBox
    currentResults="CYOD"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <input
          className="searchbox-input"
          onChange={[Function]}
          placeholder="CYMX CYUL..."
          type="text"
          value=""
        />,
        <withRouter(LinkButton)
          className="searchbox-button"
          onClick={[Function]}
          to={
            Object {
              "pathname": "/airports",
              "search": "?stations=",
            }
          }
        >
          Get Info
        </withRouter(LinkButton)>,
      ],
      "className": "searchbox",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "searchbox-input",
          "onChange": [Function],
          "placeholder": "CYMX CYUL...",
          "type": "text",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": "Get Info",
          "className": "searchbox-button",
          "onClick": [Function],
          "to": Object {
            "pathname": "/airports",
            "search": "?stations=",
          },
        },
        "ref": null,
        "rendered": "Get Info",
        "type": [Function],
      },
    ],
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <input
            className="searchbox-input"
            onChange={[Function]}
            placeholder="CYMX CYUL..."
            type="text"
            value=""
          />,
          <withRouter(LinkButton)
            className="searchbox-button"
            onClick={[Function]}
            to={
              Object {
                "pathname": "/airports",
                "search": "?stations=",
              }
            }
          >
            Get Info
          </withRouter(LinkButton)>,
        ],
        "className": "searchbox",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "searchbox-input",
            "onChange": [Function],
            "placeholder": "CYMX CYUL...",
            "type": "text",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "Get Info",
            "className": "searchbox-button",
            "onClick": [Function],
            "to": Object {
              "pathname": "/airports",
              "search": "?stations=",
            },
          },
          "ref": null,
          "rendered": "Get Info",
          "type": [Function],
        },
      ],
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
